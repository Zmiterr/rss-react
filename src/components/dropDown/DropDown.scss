.sort {
	margin-top: 15px;
}
