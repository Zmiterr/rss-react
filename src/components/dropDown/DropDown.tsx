import React, { ReactElement } from 'react';
import { IDropDownProps } from '../../types/types';

import './DropDown.scss';

const DropDown = ({ sortBy, setSortBy }: IDropDownProps): ReactElement => {
	return (
		<div className="sort">
			<label htmlFor="sortBy">
				Sort by:
				<select
					className="form__select"
					name="sortBy"
					onChange={(e) => setSortBy(e.target.value)}
					value={sortBy}
				>
					<option value="relevancy">Relevancy</option>
					<option value="popularity">Popularity</option>
					<option value="publishedAt">Date</option>
				</select>
			</label>
		</div>
	);
};

export default DropDown;
