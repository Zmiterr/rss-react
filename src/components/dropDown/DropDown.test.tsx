import React from "react";
import DropDown from './DropDown';
//test deps
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { BrowserRouter } from 'react-router-dom';

Enzyme.configure({ adapter: new Adapter() })


describe("DropDown component", () => {
	it("should render DropDown component", () => {
		const component = shallow(<BrowserRouter><DropDown  sortBy={'Relevancy'}/></BrowserRouter>);
		expect(component).toMatchSnapshot();
	});
});

