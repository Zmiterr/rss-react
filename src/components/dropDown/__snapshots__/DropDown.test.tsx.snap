// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown component should render DropDown component 1`] = `ShallowWrapper {}`;

exports[`Form component should render Form component 1`] = `ShallowWrapper {}`;
